src/main/python/SearchWidget.py

```diff
@@ -0,0 +1,168 @@
+            table_text = self._build_table_text(e.get_content())
+            if table_text:
+                self.result_found.emit([e.formatted_date, table_text])
+
+        self.finished.emit()
+        self._is_running = False
+
+    def _build_table_text(self, entryText):
+        match_positions = [m.start() for m in re.finditer(self._searchText, entryText, re.IGNORECASE)]
+
+        if len(match_positions) == 0:
+            return ""
+
+        pos = match_positions[0]
+
+        end = pos + 78
+        if end >= len(entryText):
+            table_text = entryText[pos:]
+        else:
+            table_text = entryText[pos:end] + "..."
+
+        if len(match_positions) > 1:
+            table_text = table_text + "(and " + str(len(match_positions)-1) + " more)"
+
+        return table_text
+
+
+class SearchWidget(QtWidgets.QWidget):
+    def __init__(self, edit_pane: EditPane, web_view: WebView):
+        super().__init__()
+
+        self.edit_pane = edit_pane
+        self.web_view = web_view
+        self.worker = None
+
+        self.searchText = QtWidgets.QLineEdit(self)
+        self.searchText.returnPressed.connect(self.search_clicked)
+
+        self.searchButton = QtWidgets.QPushButton("Search", self)
+        self.searchButton.clicked.connect(self.search_clicked)
+
+        toolbar = QtWidgets.QToolBar(self)
+        toolbar.addWidget(self.searchText)
+        toolbar.addWidget(self.searchButton)
+
+        self.progress_bar = QtWidgets.QProgressBar()
+        self.progress_bar.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
+        self.progress_bar.setVisible(False)
+
+        self.table = QtWidgets.QTableWidget(self)
+        self.table.setColumnCount(2)
+        self.table.setHorizontalHeaderLabels(["Date", "Text"])
+        self.table.resizeRowsToContents()
+        self.table.verticalHeader().setVisible(False)
+        self.table.setSelectionBehavior(QtWidgets.QTableView.SelectionBehavior.SelectRows)
+        self.table.setSelectionMode(self.table.SelectionMode.SingleSelection)
+        self.table.setEditTriggers(QtWidgets.QTableWidget.EditTrigger.NoEditTriggers)
+        self.table.itemSelectionChanged.connect(self.on_row_selected)
+        self.table.horizontalHeader().setSortIndicatorShown(True)
+        self.table.setSortingEnabled(True)
+        self.table.setSizeAdjustPolicy(QtWidgets.QAbstractScrollArea.SizeAdjustPolicy.AdjustToContentsOnFirstShow)
+        self.table.horizontalHeader().setStretchLastSection(True)
+
+        layout = QtWidgets.QVBoxLayout()
+        layout.addWidget(toolbar)
+        layout.addWidget(self.progress_bar)
+        layout.addWidget(self.table)
+        self.setLayout(layout)
+
+
+    def search_clicked(self):
+        if self.worker and self.worker.is_running():
+            self.worker.interrupt()
+            return
+
+        self.progress_bar.setValue(0)
+        self.progress_bar.setVisible(True)
+
+        if self.searchText.text() == "":
+            return

```

Comment:
It would be best if the check for 
```python
if self.searchText.text() == ""
``` 
is above 

```python
self.progress_bar.setValue(0)
self.progress_bar.setVisible(True)
```

Because otherwise an empty search leaves a visible, empty progress bar.

Reply:
Good point. Fixed this as you suggested.